### PR TITLE
fix: update JWT cookie at every request

### LINKED ISSUE #18880: Support for stateless security in Flow
### Describe your motivation

This is a spike issue aiming to figure out what is needed in Flow to support stateless security (authentication) in a way that consistent with the [Hilla's stateless security](https://hilla.dev/docs/react/guides/security/spring-stateless).

### Describe the solution you'd like

The server-side part should be ready, see JWT auth API in [VaadinWebSecurity](https://github.com/vaadin/flow/blob/725c7ad5a836f553f93d31bac992102398dfbb4d/vaadin-spring/src/main/java/com/vaadin/flow/spring/security/VaadinWebSecurity.java#L552), which is used by Hilla. Flow would need the counter part in Flow client.

### Additional context

Vaadin 24.4.

### PR DESCRIPTION
**Pull request by mcollovati**
## Description

Fixes stateless authentication, by updating the JWT cookies at every request, preventing the browser to remove the cookie after initial max-age time is expired.

Fixes https://github.com/vaadin/flow/issues/18895

## Type of change

- [x] Bugfix
- [ ] Feature

## Checklist

- [x] I have read the contribution guide: https://vaadin.com/docs/latest/guide/contributing/overview/
- [x] I have added a description following the guideline.
- [x] The issue is created in the corresponding repository and I have referenced it.
- [x] I have added tests to ensure my change is effective and works as intended.
- [x] New and existing tests are passing locally with my change.
- [ ] I have performed self-review and corrected misspellings.

#### Additional for `Feature` type of change

- [ ] Enhancement / new feature was discussed in a corresponding GitHub issue and Acceptance Criteria were created.

### CODE REVIEW

**Comment by mcollovati** [comment]
Tested the change also with Hilla integration test

**Comment by mcollovati** [comment]
Removed workaround for https://github.com/spring-projects/spring-security/issues/12579 since it has been fixed.

File: vaadin-spring/src/main/java/com/vaadin/flow/spring/security/stateless/VaadinStatelessSecurityConfigurer.java [hunk 1]
```diff
@@ -84,18 +97,32 @@ public final class VaadinStatelessSecurityConfigurer<H extends HttpSecurityBuild
 
     private SecretKeyConfigurer secretKeyConfigurer;
 
-    public void setSharedObjects(HttpSecurity http) {
+    public static void apply(HttpSecurity http,
```
Comment (mshabarov) [inline vaadin-spring/src/main/java/com/vaadin/flow/spring/security/stateless/VaadinStatelessSecurityConfigurer.java:100]: Needs javadocs

**Review by mshabarov — CHANGES_REQUESTED** [review]

**Comment by mshabarov** [comment]
Tested the `hilla-auth` example project with this patch and the expiration date of cookie and saved JWT token are changed from request to request as expected.

File: vaadin-spring/src/main/java/com/vaadin/flow/spring/security/stateless/VaadinStatelessSecurityConfigurer.java [hunk 2]
```diff
@@ -84,18 +97,32 @@ public final class VaadinStatelessSecurityConfigurer<H extends HttpSecurityBuild
 
     private SecretKeyConfigurer secretKeyConfigurer;
 
-    public void setSharedObjects(HttpSecurity http) {
```
[the remark this answers is not in the document]
Comment (mcollovati) [reply vaadin-spring/src/main/java/com/vaadin/flow/spring/security/stateless/VaadinStatelessSecurityConfigurer.java:87]: Make sense. I'll change it

File: vaadin-spring/src/main/java/com/vaadin/flow/spring/security/stateless/VaadinStatelessSecurityConfigurer.java (hunk 1 above)
Comment (mcollovati) [reply vaadin-spring/src/main/java/com/vaadin/flow/spring/security/stateless/VaadinStatelessSecurityConfigurer.java:100]: added

**Review by mshabarov — APPROVED** [review]

**Pull request merged**